src/_guides/language/concurrency/index.md

```diff
@@ -0,0 +1,454 @@
+O(n) in [big O notation][].
+However, when you send the data using `Isolate.exit()`,
+then the memory that holds the message in the exiting isolate isn’t copied,
+but instead is transferred to the receiving isolate.
+That transfer is quick and completes in constant time — O(1).
+
+[big O notation]: https://en.wikipedia.org/wiki/Big_O_notation
+
+{{site.alert.version-note}}
+  `Isolate.exit()` was added in 2.15.
+  Previous releases support only explicit message passing,
+  using `Isolate.send()` as shown in the next section's example.
+{{site.alert.end}}
+
+The following figure illustrates the communication between
+the main isolate and the worker isolate:
+
+![A figure showing the previous snippets of code running in the main isolate and in the worker isolate](/guides/language/concurrency/images/isolate-api.png)
+
+
+### Sending multiple messages between isolates
+
+If you need more communication between isolates,
+then you need to use the [`send()` method][] of `SendPort`.
+One common pattern, which the following figure shows,
+is for the main isolate to send a request message to the worker isolate,
+which then sends one or more reply messages.
+
+![A figure showing the main isolate spawning the isolate and then sending a request message, which the worker isolate responds to with a reply message; two request-reply cycles are shown](/guides/language/concurrency/images/isolate-custom-bg-worker.png)
+
+For examples of sending multiple messages,
+see the following [isolate samples][]:
+
+* [send_and_receive.dart][],
+  which shows how to send a message from
+  the main isolate to the spawned isolate.
+  It’s otherwise similar to the preceding example.
+* [long_running_isolate.dart][],
+  which shows how to spawn a long-running isolate that
+  receives and sends multiple times.
+
+{% assign samples = "https://github.com/dart-lang/samples/tree/master/isolates" %}
+
+[isolate samples]: {{ samples }}
+[send_and_receive.dart]: {{ samples }}/bin/send_and_receive.dart
+[long_running_isolate.dart]: {{ samples }}/bin/long_running_isolate.dart
+
+{{site.alert.tip}}
+  The `send()` method can be slow.
+  When possible,
+  avoid using `send()` to send large amounts of data.
+  Instead, send the data when the worker isolate exits,
+  as described in the previous section.
+{{site.alert.end}}
+
+## Performance and isolate groups
+
+When an isolate calls [`Isolate.spawn()`][],
+the two isolates have the same executable code
+and are in the same _isolate group_.
+Isolate groups enable performance optimizations such as sharing code;
+a new isolate immediately runs the code owned by the isolate group.
+Also, `Isolate.exit()` works only when the isolates
+are in the same isolate group.
+
+In some special cases,
+you might need to use [`Isolate.spawnUri()`][],
+which sets up the new isolate with a copy of the code
+that's at the specified URI.
+However, `spawnUri()` is much slower than `spawn()`,
+and the new isolate isn't in its spawner's isolate group.
```

Comment:
Might be worth mentioning that `Isolate.spawnUri` is not supported on Flutter.

Reply:
I had no idea! Will do.